straxen/plugins/pulse_processing.py

```diff
@@ -341,7 +343,13 @@ def software_he_veto(records, to_pe, chunk_end,
     rough_sum(regions, records, to_pe, veto_n, veto_res)
     regions['data'] /= np.max(regions['data'], axis=1)[:, np.newaxis]
 
-    pass_veto = strax.find_hits(regions, min_amplitude=pass_veto_fraction)
+    # This will ensure the actual peaks get passed, but the stuff around them
+    # Like missplit peaks, will not.
+    # This should be used if trying to keep some peaks you'd otherwise cut.
+    if fractional_pass > 0 and np.random.choice(2, 1, p=[fractional_pass, 1 - fractional_pass]) == 0:
```

Comment:
Could you clarify this part a bit. Maybe give an example, I do not understand what you are doing here 😆

Reply:
Hey, yes so imagine that we want to save 10% of peaks that pass the veto (just so that we can still have _some_ statistics in the case of, say a veto with an infinite pass fraction.)
Then, I randomly flip a coin with a binomial probability determined by the fraction I select (in this case 0.1 chance of it being heads up.) 
If it is heads up for a given chunk, then I just pass the original veto-triggering peaks (not their tails though, that's what the 0.9 is doing in the first if case for pass fraction, making sure we only pass the original HE peak.) 
If not, then I just veto as normal (so this is something you'll only use in cases where your 'tail_veto_pass_fraction' is above 1, otherwise it doesn't really make any sense.) 

My notebook [here](https://github.com/XENONnT/analysiscode/blob/master/Fast_Response_Analysis/shev_lxe/lxe_shev_plugin.ipynb) shows this new setting varied between 0 and 0.1